packages/@react-native-windows/codegen/Document.md

```diff
@@ -0,0 +1,386 @@
+# React Native TurboModule C++ Code Generation
+
+## Running Codegen
+
+`@react-native-windows/codegen` only supports reading turbo modules in Flow today.
+
+`react-native-windows-codegen --file <FILE> --files <FILES> --namespace <NAMESPACE> --libraryName <LIBRARY>`
+
+- `file`: An input file.
+- `files`: a glob pattern for multiple input files.
+- `namespace`: The namespace for the generated spec file. For example: `Microsoft::ReactNativeSpecs`.
+- `libraryName`: part of the output folder for components.
+
+After successfully executed the codegen,
+a C++ spec class will be generated in a header file.
+
+## TypeScript TurboModule Syntax
+
+A TypeScript TurboModule definition file must contain:
+
+- Exactly one module spec interface inheriting from `TurboModule`.
+- An export statement in this format `export default TurboModuleRegistry.getEnforcing<SPEC-INTERFACE-TYPE>('MODULE-NAME');`
+
+```TypeScript
+import {Int32, TurboModule, TurboModuleRegistry} from `react-native`;
```

Comment:
>Int32 [](http://example.com/codeflow?start=8&length=5)

Do we really have Int32 defined in 'react-native'?

Reply:
No. `Int32` is used in the flow front-end. For the `react-native-tscodegen`, you just need to import an alias of `number` as `Int32` from any file (the alias cannot be defined in the file containing a turbo module), it will just work.

Reply:
The, please change the example where it imports the Int32 from the react-native package. We must not mislead developers with wrong examples.